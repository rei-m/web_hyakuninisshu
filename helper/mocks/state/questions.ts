import { questionsTypes } from '@src/state/questions';

export const MOCK_YOMIFUDA_1: questionsTypes.YomiFuda = {
  karutaNo: 1,
  shoku: '秋の田の',
  niku: 'かりほの庵の',
  sanku: '苫をあらみ',
};

export const MOCK_TORIFUDA_1: questionsTypes.ToriFuda = {
  karutaNo: 1,
  shiku: 'きりにぬれつつ',
  kekku: 'わがころもでは',
};

export const MOCK_TORIFUDA_2: questionsTypes.ToriFuda = {
  karutaNo: 2,
  shiku: 'ころもほすてふ',
  kekku: 'あまのかぐやま',
};

export const MOCK_TORIFUDA_3: questionsTypes.ToriFuda = {
  karutaNo: 3,
  shiku: 'ながながしよを',
  kekku: 'ひとりかもねむ',
};

export const MOCK_TORIFUDA_4: questionsTypes.ToriFuda = {
  karutaNo: 4,
  shiku: 'ふじのたかねに',
  kekku: 'ゆきはふりつつ',
};
