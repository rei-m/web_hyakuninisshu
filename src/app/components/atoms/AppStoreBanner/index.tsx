/* eslint-disable @next/next/no-img-element */
import Image from 'next/image';
import Box from '@mui/material/Box';

export type AppStoreBannerProps = {
  type?: 'normal' | 'reader';
};

const ASSETS = {
  normal: {
    appName: '百人一首 簡単に暗記',
    iconSrc: '/logo/ios-app-icon.png',
    storeUrl: 'https://apple.co/3BTApF8',
  },
  reader: {
    appName: '百人一首 読み上げで暗記',
    iconSrc: '/logo/ios-app-reader-icon.png',
    storeUrl: 'https://apple.co/4jcavNG',
  },
} as const;

const AppStoreBanner = ({ type = 'normal' }: AppStoreBannerProps) => (
  <Box sx={{ width: 140, display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'start' }}>
    <a href={ASSETS[type].storeUrl} target="_blank" rel="noreferrer">
      <Image src={ASSETS[type].iconSrc} alt={ASSETS[type].appName} width={100} height={100} priority />
    </a>
    <a href={ASSETS[type].storeUrl} style={{ display: 'inline-block', marginTop: 6 }}>
      <img
        src="/logo/ios-download-black.svg"
        alt="App Storeでダウンロード"
        height={40}
        style={{ verticalAlign: 'middle', objectFit: 'contain' }}
      />
    </a>
  </Box>
);

export default AppStoreBanner;
