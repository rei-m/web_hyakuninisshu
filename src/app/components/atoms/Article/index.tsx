import Box from '@mui/material/Box';

import type { HeadingProps } from '@/app/components/atoms/Heading';
import type { SxAppProps } from '@/theme';

export type ArticleProps = {
  children: React.ReactNode;
  heading: React.ReactElement<HeadingProps>;
  sx?: SxAppProps;
};

export const Article = ({ children, heading, sx }: ArticleProps) => (
  <Box component={'article'} sx={sx}>
    {heading}
    {children}
  </Box>
);
