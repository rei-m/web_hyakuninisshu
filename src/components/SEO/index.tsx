import * as React from 'react';
import Helmet from 'react-helmet';
import { graphql, StaticQuery } from 'gatsby';
import { SiteMetaData } from '@src/types';

export interface Props {
  description?: string;
  lang?: string;
  meta?: Array<{
    name: string;
    content: string;
  }>;
  keywords?: string[];
  title: string;
}

export interface DefaultSEOQueryData {
  site: {
    siteMetadata: SiteMetaData;
  };
  ogpImage: {
    publicURL: string;
  };
}

const SEO: React.FC<Props> = ({ description, lang, meta, keywords, title }) => {
  const checkedLang: string = lang ? lang : 'ja';
  const checkedKeywords: string[] = keywords ? keywords : [];
  const checkedMeta: Array<{
    name: string;
    content: string;
  }> = meta ? meta : [];

  return (
    <StaticQuery
      query={query}
      render={(data: DefaultSEOQueryData) => {
        const checkedTitle = title || data.site.siteMetadata.title;
        const checkedDescription = description || data.site.siteMetadata.description;
        return (
          <Helmet
            htmlAttributes={{
              lang: checkedLang,
            }}
            title={checkedTitle}
            meta={[
              {
                name: `description`,
                content: checkedDescription,
              },
              {
                property: `og:title`,
                content: checkedTitle,
              },
              {
                property: `og:description`,
                content: checkedDescription,
              },
              {
                property: `og:type`,
                content: `website`,
              },
              {
                property: `og:image`,
                content: `https://hyakuninanki.net${data.ogpImage.publicURL}`,
              },
              {
                name: `twitter:card`,
                content: `summary`,
              },
              {
                name: `twitter:creator`,
                content: data.site.siteMetadata.author,
              },
              {
                name: `twitter:title`,
                content: checkedTitle,
              },
              {
                name: `twitter:description`,
                content: checkedDescription,
              },
              {
                name: `twitter:image`,
                content: `https://hyakuninanki.net${data.ogpImage.publicURL}`,
              },
            ]
              .concat(
                checkedKeywords.length > 0
                  ? {
                      name: `keywords`,
                      content: checkedKeywords.join(`, `),
                    }
                  : []
              )
              .concat(checkedMeta)}
          />
        );
      }}
    />
  );
};

export default SEO;

const query = graphql`
  query DefaultSEOQuery {
    site {
      siteMetadata {
        title
        description
        author
      }
    }
    ogpImage: file(relativePath: { eq: "app-icon.png" }) {
      publicURL
    }
  }
`;
