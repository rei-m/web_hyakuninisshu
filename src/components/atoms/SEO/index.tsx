import React from 'react';
import Helmet from 'react-helmet';
import { useSEO } from '@src/hooks/staticQueries/useSEO';

export type Props = {
  description?: string;
  lang?: string;
  meta?: Array<{
    name: string;
    content: string;
  }>;
  keywords?: string[];
  title: string;
};

const SEO = ({ description, lang = 'ja', meta = [], keywords = [], title }: Props) => {
  const { site, ogpImage } = useSEO();
  const checkedTitle = title || site.siteMetadata.title;
  const checkedDescription = description || site.siteMetadata.description;
  return (
    <Helmet
      htmlAttributes={{
        lang,
      }}
      title={checkedTitle}
      meta={[
        {
          name: `description`,
          content: checkedDescription,
        },
        {
          property: `og:title`,
          content: checkedTitle,
        },
        {
          property: `og:description`,
          content: checkedDescription,
        },
        {
          property: `og:type`,
          content: `website`,
        },
        {
          property: `og:image`,
          content: `https://hyakuninanki.net${ogpImage.publicURL}`,
        },
        {
          name: `twitter:card`,
          content: `summary`,
        },
        {
          name: `twitter:creator`,
          content: site.siteMetadata.author,
        },
        {
          name: `twitter:title`,
          content: checkedTitle,
        },
        {
          name: `twitter:description`,
          content: checkedDescription,
        },
        {
          name: `twitter:image`,
          content: `https://hyakuninanki.net${ogpImage.publicURL}`,
        },
      ]
        .concat(
          keywords.length > 0
            ? {
                name: `keywords`,
                content: keywords.join(`, `),
              }
            : []
        )
        .concat(meta)}
    />
  );
};

export default SEO;
