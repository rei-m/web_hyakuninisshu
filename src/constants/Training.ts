import { OptionItem } from '../types';

export const RANGE_FROM: Array<OptionItem<number>> = [
  {
    name: '一番',
    value: 1
  },
  {
    name: '十一番',
    value: 11
  },
  {
    name: '二十一番',
    value: 21
  },
  {
    name: '三十一番',
    value: 31
  },
  {
    name: '四十一番',
    value: 41
  },
  {
    name: '五十一番',
    value: 51
  },
  {
    name: '六十一番',
    value: 61
  },
  {
    name: '七十一番',
    value: 71
  },
  {
    name: '八十一番',
    value: 81
  },
  {
    name: '九十一番',
    value: 91
  }
];

export const RANGE_TO: Array<OptionItem<number>> = [
  {
    name: '十番',
    value: 10
  },
  {
    name: '二十番',
    value: 20
  },
  {
    name: '三十番',
    value: 30
  },
  {
    name: '四十番',
    value: 40
  },
  {
    name: '五十番',
    value: 50
  },
  {
    name: '六十番',
    value: 60
  },
  {
    name: '七十番',
    value: 70
  },
  {
    name: '八十番',
    value: 80
  },
  {
    name: '九十番',
    value: 90
  },
  {
    name: '百番',
    value: 100
  }
];
