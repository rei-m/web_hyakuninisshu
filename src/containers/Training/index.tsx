import * as React from 'react';
import { connect } from 'react-redux';
import { branch, renderComponent } from 'recompose';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { Question } from '../../types';
import { GlobalState } from '../../reducers/index';
import TrainingSection from '../../components/TrainingSection';
import TrainingInitializer from '../TrainingInitializer';

export interface TrainingProps {
  started: boolean;
  question?: Question;
}

const mapStateToProps = (
  state: GlobalState,
  props: RouteComponentProps<{}>
): TrainingProps => {
  const { submitTime } = props.location.state;
  const { lastStartedTime, questions } = state.trainings;
  return {
    question: questions[0],
    started: !!lastStartedTime && lastStartedTime > submitTime
  };
};

const isStarted = ({ started }: TrainingProps) => started;

const withStartedCheck = branch<TrainingProps>(
  isStarted,
  component => component,
  _ => TrainingInitializer
);

const hasQuestion = ({ question }: TrainingProps) => !!question;

const EmptyMessage = () => <h3>指定した条件の歌はありませんでした</h3>;

const withHasQuestionCheck = branch<TrainingProps>(
  hasQuestion,
  component => component,
  renderComponent(EmptyMessage)
);

export default withRouter(
  connect(mapStateToProps)(
    withStartedCheck(withHasQuestionCheck(TrainingSection))
  )
);
