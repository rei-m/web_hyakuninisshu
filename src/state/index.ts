import { applyMiddleware, combineReducers, createStore as reduxCreateStore, Reducer, Store } from 'redux';
import thunk from 'redux-thunk';
import { questions, QuestionsState } from '@src/state/questions';
import { ui, UiState } from '@src/state/ui';

export interface GlobalState {
  questions: QuestionsState;
  ui: UiState;
}

export const reducer: Reducer<GlobalState> = combineReducers<GlobalState>({
  questions,
  ui,
});

export const createStore = (): Store<GlobalState> => reduxCreateStore(reducer, applyMiddleware(thunk));
