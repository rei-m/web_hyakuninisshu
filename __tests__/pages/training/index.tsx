import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import { Provider } from 'react-redux';
import Training from '@src/pages/training';
import { DefaultSEOQueryData } from '@src/components/atoms/SEO';
import { createStore } from '@src/state';
import { setUpQueryOnce } from '@test/helpers';

describe('/training', () => {
  beforeEach(() => {
    setUpQueryOnce<DefaultSEOQueryData>({
      site: {
        siteMetadata: {
          title: 'training title',
          description: 'training description',
          author: '@rei-m',
        },
      },
      ogpImage: {
        publicURL: 'http://localhost:8000/app.png',
      },
    });
  });

  it('should render component', () => {
    const renderer = ReactTestRenderer.create(
      <Provider store={createStore()}>
        <Training />
      </Provider>
    );
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
