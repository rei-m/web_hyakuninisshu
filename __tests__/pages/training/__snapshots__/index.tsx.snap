// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/training should render component 1`] = `
Array [
  <header
    className="sc-EHOje iDSNuT"
  >
    <div
      className="sc-ifAKCX gDXiY"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <i
        className="material-icons sc-gzVnrw frDxzG"
        color="#fff"
        data-test="back"
        onClick={[Function]}
      >
        arrow_back
      </i>
    </div>
    <h1
      className="sc-bZQynM cSHetA"
    >
      百人一首 - 練習 -
    </h1>
  </header>,
  <div
    className="sc-gZMcBi gobYat"
  >
    <section
      className="sc-chPdSV gytgom"
    >
      <h1
        className="sc-jTzLTM fIzVOw"
      >
        出題設定
      </h1>
      <div
        className="sc-fjdhpX QChbQ"
      >
        <div
          className="sc-jzJRlG hmtsxu"
        >
          この欄は広告枠です
        </div>
      </div>
      <div
        className="sc-kgoBCf djnryy"
      >
        <form
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="MuiFormControl-root-3 Component-formControl-1"
            style={
              Object {
                "marginBottom": "16px",
              }
            }
          >
            <label
              className="sc-bwzfXH jjTplH"
            >
              出題範囲
            </label>
            <div
              className="sc-bdVaJa grbsvW"
            >
              <div
                className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-2"
                onClick={[Function]}
              >
                <div
                  className="MuiSelect-root-7"
                >
                  <div
                    aria-haspopup="true"
                    aria-pressed="false"
                    className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    一番
                  </div>
                  <input
                    id="id-range-from"
                    name="rangeFrom"
                    type="hidden"
                    value="1"
                  />
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-44 MuiSelect-icon-13"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
              <span
                className="sc-htpNat dDpdjb"
              >
                〜
              </span>
              <div
                className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-2"
                onClick={[Function]}
              >
                <div
                  className="MuiSelect-root-7"
                >
                  <div
                    aria-haspopup="true"
                    aria-pressed="false"
                    className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    百番
                  </div>
                  <input
                    id="id-range-to"
                    name="rangeTo"
                    type="hidden"
                    value="100"
                  />
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-44 MuiSelect-icon-13"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="MuiFormControl-root-3 Component-formControl-57"
            style={
              Object {
                "marginBottom": "16px",
              }
            }
          >
            <label
              className="MuiFormLabel-root-71 MuiFormLabel-filled-75 MuiInputLabel-root-60 MuiInputLabel-formControl-65 MuiInputLabel-animated-68 MuiInputLabel-shrink-67 Component-label-58"
              data-shrink={true}
              htmlFor="id-kimariji"
            >
              決まり字
            </label>
            <div
              className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-59"
              onClick={[Function]}
            >
              <div
                className="MuiSelect-root-7"
              >
                <div
                  aria-haspopup="true"
                  aria-pressed="false"
                  className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  指定しない
                </div>
                <input
                  id="id-kimariji"
                  name="kimariji"
                  type="hidden"
                  value="0"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-44 MuiSelect-icon-13"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="MuiFormControl-root-3 Component-formControl-57"
            style={
              Object {
                "marginBottom": "16px",
              }
            }
          >
            <label
              className="MuiFormLabel-root-71 MuiFormLabel-filled-75 MuiInputLabel-root-60 MuiInputLabel-formControl-65 MuiInputLabel-animated-68 MuiInputLabel-shrink-67 Component-label-58"
              data-shrink={true}
              htmlFor="id-color"
            >
              五色
            </label>
            <div
              className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-59"
              onClick={[Function]}
            >
              <div
                className="MuiSelect-root-7"
              >
                <div
                  aria-haspopup="true"
                  aria-pressed="false"
                  className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  指定しない
                </div>
                <input
                  id="id-color"
                  name="color"
                  type="hidden"
                  value="none"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-44 MuiSelect-icon-13"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="MuiFormControl-root-3 Component-formControl-57"
            style={
              Object {
                "marginBottom": "16px",
              }
            }
          >
            <label
              className="MuiFormLabel-root-71 MuiFormLabel-filled-75 MuiInputLabel-root-60 MuiInputLabel-formControl-65 MuiInputLabel-animated-68 MuiInputLabel-shrink-67 Component-label-58"
              data-shrink={true}
              htmlFor="id-kamiNoKuStyle"
            >
              上の句
            </label>
            <div
              className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-59"
              onClick={[Function]}
            >
              <div
                className="MuiSelect-root-7"
              >
                <div
                  aria-haspopup="true"
                  aria-pressed="false"
                  className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  漢字と仮名で表示
                </div>
                <input
                  id="id-kamiNoKuStyle"
                  name="kamiNoKuStyle"
                  type="hidden"
                  value="0"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-44 MuiSelect-icon-13"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            className="MuiFormControl-root-3 Component-formControl-57"
            style={
              Object {
                "marginBottom": "16px",
              }
            }
          >
            <label
              className="MuiFormLabel-root-71 MuiFormLabel-filled-75 MuiInputLabel-root-60 MuiInputLabel-formControl-65 MuiInputLabel-animated-68 MuiInputLabel-shrink-67 Component-label-58"
              data-shrink={true}
              htmlFor="id-shimoNoKuStyle"
            >
              下の句
            </label>
            <div
              className="MuiInputBase-root-27 MuiInput-root-14 MuiInput-underline-18 MuiInputBase-formControl-28 MuiInput-formControl-15 Component-select-59"
              onClick={[Function]}
            >
              <div
                className="MuiSelect-root-7"
              >
                <div
                  aria-haspopup="true"
                  aria-pressed="false"
                  className="MuiSelect-select-8 MuiSelect-selectMenu-11 MuiInputBase-input-37 MuiInput-input-22"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  すべて仮名で表示
                </div>
                <input
                  id="id-shimoNoKuStyle"
                  name="shimoNoKuStyle"
                  type="hidden"
                  value="1"
                />
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-44 MuiSelect-icon-13"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <button
            className="MuiButtonBase-root-109 MuiButton-root-83 MuiButton-outlined-91 Component-button-78 Component-primaryButton-80"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={
              Object {
                "marginTop": "32px",
              }
            }
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-84"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-112 MuiIcon-fontSizeSmall-119 Component-primaryIcon-82"
              >
                edit
              </span>
              練習をはじめる
            </span>
            <span
              className="MuiTouchRipple-root-121"
            />
          </button>
        </form>
      </div>
      <div
        className="sc-cSHVUG lopQnD"
      >
        <div
          className="sc-kAzzGY dTlPEQ"
        >
          この欄は広告枠です
        </div>
      </div>
    </section>
  </div>,
  <nav
    className="sc-dnqmqq hTWAOb"
  >
    <span
      className="sc-iwsKbI gEZdpE"
    >
      <div
        className="sc-ifAKCX gDXiY"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/training"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-htoDjs dEepJM"
          >
            create
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-iwsKbI gEZdpE"
    >
      <div
        className="sc-ifAKCX gDXiY"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/exam"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-htoDjs NoVcW"
          >
            note
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-iwsKbI gEZdpE"
    >
      <div
        className="sc-ifAKCX gDXiY"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/karutas"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-htoDjs jvjqPK"
          >
            library_books
          </i>
        </a>
      </div>
    </span>
    <span
      className="sc-iwsKbI gEZdpE"
    >
      <div
        className="sc-ifAKCX gDXiY"
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <a
          color="#fff"
          href="/about"
          style={
            Object {
              "height": "100%",
              "width": "100%",
            }
          }
        >
          <i
            className="material-icons sc-htoDjs eFmGnn"
          >
            settings
          </i>
        </a>
      </div>
    </span>
  </nav>,
]
`;
